### ISSUE TITLE
Improve Wallet coverage

### ISSUE THREAD

**Issue by t-neIson** [solana-labs-workspace/solana#21]
#### Problem
Wallet module code coverage was abysmal, and included several ignored, flaky, integration tests.

#### Summary of Changes
Add RpcRequestHandler trait in order to test WalletCommand processing with mocked data instead of Rpc i/o
Fix flaky tests and move to wallet integration tests directory
Also renamed "AirDrop" to "Airdrop" because it annoyed me

Fixes #2365

**Comment by mvilnes** [comment]
One glaring hole in this is any tests for the Deploy wallet command. I will follow up with a separate PR shortly.

**Comment by CrieasofCarrots** [comment]
So much more coverage! w00t!!

Seems like a whole lot of copy-paste going on and that with a little polish, you could make this PR a few hundred lines shorter with all the same coverage. Also, those calls to `sleep` are concerning. How long does it take to execute the unit tests? How long to execute the integration tests? The latter can be a little longer, but still seems like it'd be under a second if we can find a way to remove the sleeps.

**Comment by jeffweshington** [comment]
Tx confirmation retry code is now skipping sleeps in unit tests.
I think that @mvilnes concluded that the sleeps are necessary when interacting with a real node, so they need to stay for live clients and integration tests. But it's possible that a shorter sleep would be sufficient. Thoughts, @mvilnes ?

**Comment by mvilnes** [comment]
re: sleeps, i forget :)
Let’s remove them and see what breaks (and then de-race that code as needed)

**Comment by CrieasofCarrots** [comment]
wins this round. Can you rebase?

**Comment by jeffweshington** [comment]
Will remove sleeps in a separate PR to isolate what breaks.
Otherwise, I think this is ready

**Comment by t-neIson** [comment]
```
  test=${test##*/} # basename x
  test=${test%.rs} # basename x .rs
 _ cargo test --all --verbose --features="$FEATURES" --test="$test" -- --test-threads=1 --nocapture
```
Does this `--all` actually do anything when you also have `--test="$test"`?

**Comment by sakeridge** [comment]
Yes. Cargo complains it can’t find whatever named test in submodule/tests without it.

**Comment by mvilnes** [comment]
what do you think? At a glance, this looks like it could become painfully slow. Is adding `--all` here the right thing to do or should it be a `cd wallet &&` ?

**Comment by CrieasofCarrots** [comment]
Incidentally, my original plan was to make the match generic,`for test in tests/*.rs */tests/*.rs;`, so as to not require any upkeep for future submodule integration tests. But the tests in `ledger-tool/tests` need some attention for that to work.

**Comment by jeffweshington** [comment]
Looks great! Thanks for all the polish!
